compiler/AST/LoopExpr.cpp

```diff
@@ -732,8 +735,19 @@ static CallExpr* buildLoopExprFunctions(LoopExpr* loopExpr) {
   bool insideArgSymbol = isArgSymbol(loopExpr->parentSymbol) ||
                          isTypeSymbol(loopExpr->parentSymbol);
 
+  // The loop expression may receive additional vars via attributes applied
+  // to its variable. This is represented by enclosing the LoopExpr inside
+  // a "gpu attribute block". See if we need to handle that, as well.
+  BlockStmt* attrBlock = nullptr;
+  BlockStmt* primsFromAttrs = nullptr;
+  if ((attrBlock = findEnclosingGpuAttributeBlock(loopExpr))) {
```

Comment:
extra parentheses

Reply:
These are not extra. Since `if (x = y)` is a common typo for `if (x == y)`, C/C++ compilers require _actual_ assignment to be parenthesized an additional time in conditions.

Reply:
Oh, right.